src/shims/foreign_items/posix.rs

```diff
@@ -255,6 +253,93 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 this.write_null(dest)?;
             }
 
+            // Threading
+            "pthread_create" => {
```

Comment:
Please move this implementation to another file. This match here should only contain very short arms.